src/schema/formats.yaml

```diff
@@ -0,0 +1,66 @@
+---
+# This file defines valid patterns for different formats
+# Entity patterns
+label:
+  description: |
+    Freeform labels without special characters.
+  pattern: ^[0-9a-zA-Z]+$
+index:
+  description: |
+    Non-negative, non-zero integers, optionally prefixed with leading zeros for sortability.
+  pattern: ^[0-9]+$
+# Metadata types
+string:
+  description: |
+    A basic string type (not a specific format).
+    This should allow any free-form string *except* "n/a".
+  pattern: ^(?!(n/a)$).*$
+integer:
+  description: |
+    An integer.
+  pattern: ^[+-]?\d+$
+number:
+  description: |
+    A number.
+  pattern: ^[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)$
+boolean:
+  description: |
+    A boolean.
+  pattern: ^(true|false)$
+# String patterns
+date:
```

Comment:
`date` is included in the schema as a "pattern" rather than a "format". That was just an oversight on my part. I would fix it here, but that would cause a conflict with #883, so I will address it in that PR.

Reply:
Fixed in e9c4f5f (part of #883).